src/main/java/eu/europeana/metis/sandbox/service/debias/ProcessingState.java

```diff
@@ -42,12 +42,12 @@ public void success(Integer datasetId) {
   public boolean process(Integer datasetId) {
     LOGGER.info("{} {}", STATE_NAME, datasetId);
     try {
-      DetectionEntity detectionEntity = detectRepository.findDetectionEntityByDatasetId_DatasetId(datasetId);
-      if (detectionEntity == null) {
+      DatasetDeBiasEntity datasetDeBiasEntity = datasetDeBiasRepository.findDetectionEntityByDatasetId_DatasetId(datasetId);
+      if (datasetDeBiasEntity == null) {
         fail(datasetId);
         LOGGER.warn("invalid state {} {}", STATE_NAME, datasetId);
       } else {
-        detectRepository.updateState(datasetId, STATE_NAME);
+        datasetDeBiasRepository.updateState(datasetId, STATE_NAME);
         success(datasetId);
         LOGGER.info("success {} {}", STATE_NAME, datasetId);
         // TODO: processing logic here
```

Comment:
Is something still missing? This TODO should be evaluated.

Reply:
I will do a clean up.